test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java

```diff
@@ -225,6 +260,14 @@ public Releasable capturing(Class<?>... classes) {
                 Loggers.removeAppender(logger, this);
             }
             stop();
+            // check that all expectations have been evaluated before this is released
+            for (WrappedLoggingExpectation expectation : expectations) {
+                assertThat(
+                    "Method assertMatched() not called on LoggingExpectation instance before release: " + expectation,
+                    expectation.assertMatchedCalled.get(),
+                    is(true)
```

Comment:
should the assertAllExpectationsMatched be called here?

Reply:
I'd prefer not, in case a test wants to check expectations, then add more expectations and check again

Reply:
In particular, an `UnseenEventExpectation` will fail if it sees the event, but that might be expected to happen in between the last `mockLogAppender.assertAllExpectationsMatched()` and the end of the `capturing()` block.

Reply:
makes sense, thank you